src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -892,6 +895,42 @@ private String getNearestClassFrameName() {
         return frame.getFrameName();
     }
 
+    /**
+     * Checks if the token is a Lambda parameter.
+     * @param ast the {@code DetailAST} value of the token to be checked
+     * @return true if the token is a Lambda parameter
+     */
+    private static boolean isLambdaParameter(DetailAST ast) {
+        DetailAST parent;
+        for (parent = ast.getParent(); parent != null;
+             parent = parent.getParent()) {
+            if (parent.getType() == TokenTypes.LAMBDA) {
+                break;
+            }
+        }
+        final boolean returnValue;
+        if (parent == null) {
+            returnValue = false;
+        }
+        else if (ast.getType() == TokenTypes.PARAMETER_DEF) {
+            returnValue = true;
+        }
+        else {
+            final DetailAST lambdaParameters = parent.findFirstToken(TokenTypes.PARAMETERS);
+            if (lambdaParameters == null) {
+                returnValue = parent.getFirstChild().getText().equals(ast.getText());
```

Comment:
There is no need in comparison of the text. AST is static and generated at once. You can just use
```java
parent.getFirstChild().equals(ast)
```

The ```equals``` method will be invoked from BaseAST:

```java
    public boolean equals(AST var1) {
        return var1 == null?false:(this.getText() == null && var1.getText() != null || this.getText() != null && var1.getText() == null?false:(this.getText() == null && var1.getText() == null?this.getType() == var1.getType():this.getText().equals(var1.getText()) && this.getType() == var1.getType()));
    }
```

@romani 
Am I right?